index.js

```diff
@@ -1,253 +1,37 @@
-    propStream = require('prop-stream'),
-    settings = require('pelias-config').generate(),
-    dbclient = require('pelias-dbclient')(),
-    OsmiumStream = require('osmium-stream');
-
-var Backend = require('geopipes-elasticsearch-backend');
-
-// esclient.livestats();
-
-var backend = function( index, type ){
-  return new Backend( dbclient.client, index, type );
-};
-
-// @todo: extract this or refactor suggester to make it a stream factory
-var bun = require('bun');
-function generateSuggester(){
-  return bun([
-    suggester.streams.suggestable(),
-    suggester.streams.suggester( suggester.generators )
-  ]);
-}
-
-// use datapath setting from your config file
-var basepath = settings.imports.openstreetmap.datapath;
-var filename = settings.imports.openstreetmap.import[0].filename;
-var leveldbpath = settings.imports.openstreetmap.leveldbpath;
-
-// testing
-// basepath = '/media/hdd/osm/mapzen-metro';
-// filename = 'wellington.osm.pbf';
-
-var pbfFilePath = basepath + '/' + filename;
-// check pbf file exists
-try {
-  fs.statSync( pbfFilePath );
-} catch( e ){
-  console.error( 'failed to load pbf file:', pbfFilePath );
-  process.exit(1);
-}
-
-// check leveldb dir exists
-try {
-  fs.statSync( leveldbpath );
-} catch( e ){
-  console.error( 'failed to open:', leveldbpath );
-  process.exit(1);
-}
-
-// generic streams
-var stringify =                require('./stream/stringify');
-var devnull =                  require('./stream/devnull');
-var required =                 require('./stream/required');
-// var sporadic =                 require('./stream/sporadic');
+/**
+  The openstreetmap importer provides a way of parsing, mapping and
+  augmenting OSM data in to elasticsearch.
+**/
 
-// sinks
-var backend = {
-  es: {
-    osmnode:                  backend('pelias', 'osmnode'),
-    osmway:                   backend('pelias', 'osmway'),
-    geonames:                 backend('pelias', 'geoname'),
-    admin0:                   backend('pelias', 'admin0'),
-    admin1:                   backend('pelias', 'admin1'),
-    admin2:                   backend('pelias', 'admin2'),
-    local_admin:              backend('pelias', 'local_admin'),
-    locality:                 backend('pelias', 'locality'),
-    neighborhood:             backend('pelias', 'neighborhood')
-  },
-  level: {
-    osmnodecentroids:         multilevel( levelup( leveldbpath ), 'm' )
-  }
-};
-
-// forkers
-var osm_types =                require('./stream/osm_types');
-
-// osm
-var osm = {
-  way: {
-    denormalizer:              require('./stream/osm/way/denormalizer'),
-  },
-  any: {
-    hierarchyLookup:            require('./stream/osm/any/hierarchyLookup')
-  }
+var elasticsearch = require('pelias-dbclient'),
+    adminLookup = require('pelias-admin-lookup'),
+    suggester = require('pelias-suggester-pipeline'),
+    dbmapper = require('./stream/dbmapper');
+
+var osm = { pbf: {}, doc: {}, address: {}, tag: {} };
```

Comment:
A 'class' in JS represents an instantiable construct for which each instance may hold state for it's individual properties while sharing methods from a prototype.

This functionality is not required in this case and so we would end up exporting a singleton with no individual state or prototypal methods.

The current implementation is equivalent and forwards compatible if (hypothetically) we decided to do export a singleton in the future.

My biggest concern here is a fundamental misunderstanding of the architecture, the system is designed in a way that allows individual components to be easily added/removed from the import pipeline by simply excluding/including additional streams to the pipeline.

A good analogy is this:

```
#! /bin/bash
$> osm_parser | admin_lookup | elasticsearch;
```

if another developer would like to decompose that pipeline and fork an existing stream or add a new one then they can simply re-compose the pipeline as such:

 ```
#! /bin/bash
$> osm_parser | my_fork_of_admin_lookup | something_else | elasticsearch;
```

If we move towards a more traditional OOP architecture then we would potentially lose the advantages we gained from the unix pipes style interfaces in that interoperability and composability are sacrificed in favour of readability for traditional OOP developers.

As it currently stands this repository is completely functional and stateless, I would prefer to keep it that way.

I think this specific discussion would be best continued outside of this PR.